packages/app/pages/submission/new.tsx

```diff
@@ -1,5 +1,137 @@
+import { styled } from "@/stitches.config";
+import debounce from "lodash.debounce";
+import { useState } from "react";
+import {useRouter} from "next/router";
+import { 
+  Button, 
+  Input,
+  Text 
+} from "@cabindao/topo";
+import { useStore } from "@/store/store";
+import Title from "@/components/Title";
+import Post from "@/components/Post";
+import { IPost } from "@/interfaces";
+
+/** Styled components */
+const URLInput = styled(Input, {
+  flex: "1",
+  width: "100%"
+});
+
+const SubmitButton = styled(Button, {
+  marginTop: "$5"
+});
+
+/** Page: 'new' */
 const NewSubmission = () => {
-  return <div>new submission page</div>;
+  const router = useRouter();
+  const { currentProfile } = useStore();
+
+  const [error, setError] = useState<string>('');
+  // TODO: Use real wallet address
+  const [postData, setPostData] = useState<IPost>({
+    _id: '',
+    title: '',
+    domainText: '',
+    url: '',
+    // postedBy: currentProfile.walletAddress,
```

Comment:
it's possible for the currentProfile to be an empty object or not have an address so you'll want to check that  and redirect if there isn't one or it's possible the store isn't initialized with the current profile at this point? cc @delightfulabyss?

Reply:
I'm signed in with Metamask, so I think it should be available? In any case, a user has to be logged in to view the submission page so I think we can assume it's there?